arangod/RestHandler/RestEdgesHandler.cpp

```diff
@@ -172,8 +177,20 @@ bool RestEdgesHandler::readEdges() {
     return false;
   }
 
-  auto queryResult =
-      ::queryEdges(_vocbase, collectionName, direction, startVertex);
+  // Check if dirty reads are allowed:
+  found = false;
+  std::string const& val =
+      _request->header(StaticStrings::AllowDirtyReads, found);
+  bool allowDirtyReads = false;
+  if (found && StringUtils::boolean(val)) {
+    // This will be used in `createTransaction` below, if that creates
+    // a new transaction. Otherwise, we use the default given by the
+    // existing transaction.
+    allowDirtyReads = true;
+  }
```

Comment:
```suggestion
  // This will be used in `createTransaction` below, if that creates
  // a new transaction. Otherwise, we use the default given by the
  // existing transaction.
  allowDirtyReads = _request->parsedValue(StaticStrings::AllowDirtyReads, /*default*/ false);
```